### PR TITLE
Mkdir

### PR DESCRIPTION
**Pull request by jacobherrington**
wip do not merge

open to suggestions/help

### CODE REVIEW

File: lib/packages/mkdir.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require "set"
```
Comment (faraazahmad) [inline lib/packages/mkdir.rb]: I guess we'll start following this rule from now on. single quotes if interpolation isn't required

File: lib/packages/mkdir.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+require_relative "core"
+
+class Mkdir 
+
+  def self.main params
+    flags = params.flat_map{ |s| s.scan(/--?([^=\s]+)(?:=(\S+))?/) }
```
Comment (faraazahmad) [inline lib/packages/mkdir.rb]: what does this statement do? You'll need to document the code.

File: lib/packages/mkdir.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    params.each do |directory_name|
+      Dir.mkdir(directory_name) unless File.exists?(directory_name)
```
Comment (faraazahmad) [inline lib/packages/mkdir.rb]: use ```File.exist?``` instead

**Comment by jacobherrington** [comment]
Closing this for now, haven't had a chance to revisit it

**Pull request closed without merging**